packages/clay-card/src/CardWithFile.tsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayCard from './Card';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClaySticker from '@clayui/sticker';
+import React, {ImgHTMLAttributes} from 'react';
+import {ClayDropDownWithBasicItems} from '@clayui/drop-down';
+
+interface IProps {
+	/**
+	 * List of actions in the dropdown menu
+	 */
+	actions?: React.ComponentProps<typeof ClayDropDownWithBasicItems>['items'];
+
+	/**
+	 * Description of the file
+	 */
+	description?: React.ReactText;
+
+	/**
+	 * Flag to indicate if `aspect-ratio-item-flush` class should be
+	 * applied to the image.
+	 */
+	flushHorizontal?: boolean;
+
+	/**
+	 * Flag to indicate if `aspect-ratio-item-vertical-flush` class should be
+	 * applied to the image.
+	 */
+	flushVertical?: boolean;
+
+	/**
+	 * Path or URL to file
+	 */
+	href?: string;
+
+	/**
+	 * Object of props for `<img />` or string path to image
+	 */
+	imgProps?: React.ImgHTMLAttributes<HTMLImageElement> | string;
+
+	/**
+	 * List of labels that are applied to the file
+	 */
+	labels?: Array<
+		React.ComponentProps<typeof ClayLabel> & {value: React.ReactText}
+	>;
+
+	/**
+	 * Callback for when item is selected
+	 */
+	onSelectChange?: (val: boolean) => void;
+
+	/**
+	 * Flag to indicate if card is selected
+	 */
+	selected?: boolean;
+
+	/**
+	 * Path to clay icon spritemap
+	 */
+	spritemap: string;
+
+	/**
+	 * Values used in displaying bottom-left icon
+	 */
+	stickerProps?: {
+		content: React.ReactNode;
+		displayType: React.ComponentProps<typeof ClaySticker>['displayType'];
+	};
+
+	/**
+	 * Name of icon
+	 */
+	symbol?: string;
+
+	/**
+	 * Name of the file
+	 */
+	title: string;
+}
+
+export const ClayCardWithFile: React.FunctionComponent<IProps> = ({
+	actions,
+	description,
+	flushHorizontal,
+	flushVertical,
+	href,
+	imgProps,
+	labels,
+	onSelectChange,
+	selected = false,
+	spritemap,
+	stickerProps,
+	symbol = 'documents-and-media',
+	title,
+}) => {
+	const headerContent = (
+		<ClayCard.AspectRatio className="card-item-first">
+			{!imgProps && (
+				<div className="aspect-ratio-item aspect-ratio-item-center-middle aspect-ratio-item-fluid card-type-asset-icon">
+					<ClayIcon spritemap={spritemap} symbol={symbol} />
+				</div>
+			)}
+
+			{imgProps && (
+				<img
```

Comment:
I'm not sure `CardWithFile` will accept images in case it should only accept icons, but I understand that in the end it looks like `CardWithImage` maybe we can leave a more generic name to merge the two into one.

Reply:
hmm, yeah probably smart to make the name more generic.

`ClayCardWithBasicInfo`
or
`ClayCardWithEntity` (kinda weird cause we already have *WithUser)

We could try to merge this with `ClayCardWithUser` and make it `ClayCardWithEntity`. Not sure if we want to keep the distinction though.

Any other names come to mind?

Reply:
Hmm I don't know if I like `ClayCardWithEntity` does not seem so obvious to me its use, maybe going with `ClayCardWithInfo` it is not quite basic so in fact we can remove the word `Basic`.